Update OnePlus 6-series instructions for Android 14

Terminator-J opened the pull request:
Updated to match current template, reduced reliance on markdown tricks.

If possible, would love to see how it looks without any additional editing, and/or notes on what syntax to change in my own file revisions to keep Gabriel happy. 😁

Terminator-J committed:
Update OnePlus 6-series instructions for Android 14.0

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
 
-- A computer with current AOSP "platform-tools" installed or otherwise on your $PATH (the real ones from developer.android.com, not some script-kiddie's all-in-one bundle that's hopelessly out of date), OnePlus USB drivers installed (works for adb and OEM bootloader _fastboot_ commands), and Google's universal ADB drivers installed (needed for fastbootd mode within recovery). Linux and MacOS user ...
+- Computer:
+  - You need a computer with current AOSP "platform-tools" installed or otherwise on your $PATH (the current official ones from developer.android.com, not some script-kiddie's sketchy all-in-one bundle or your distro's packages that are two years out of date), OnePlus USB drivers installed (works for adb and OEM bootloader `fastboot` commands), and Google's universal ADB drivers installed (needed ...
```
don't add ` to fastboot command 
will render on new page with copy
for bold and italic (maybe), can use 3x*

Terminator-J replied:
I know it makes it copyable.
Fix the markdown style so it doesn't overlay extra stuff and add to visual clutter; I want some commands to be able to be styled as commands so they can be copy/pasted.

gwolf2u replied:
ok but it will render as copy on new line not as you see on github 
is like command text box with copy button on right

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
-### Optional files
+**Required files**
+
+- crDroid files for your device can be found **[here](https://sourceforge.net/projects/crdroid/files/enchilada/10.x/)**.
```
link is on download page, so just state that as it was before
there's no need to link it here also since this can change maybe and forget to change here also

Terminator-J replied:
Download page does not have buttons to download all required files, just zip and boot.img.

For first install, users need boot.img, dtbo.img, super_empty.img, vbmeta.img, and the ROM zip as a matched set.
So I'm trying to figure out the best way to quickly link "here's where you can get all the required files" with minimal extra words.

This is a packaging infrastructure issue I'm trying to work around.

Will update instructions to say "Click on 'Download older versions' button on download page to show all files" instead, and remove link.

But that's MORE words, not less. Not as clean. 😅

gwolf2u replied:
in this case, link to recovery dir or a different dir and state is not crDroid files needed, but rather specific images
if need to link to crdroid.zip, just state dl page
that's the idea
maybe use a firmware flash script and link to it like I do to guacamole/lemonade(p) - see how I did it

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+
+- You'll need a complete set of the following files:
+
+  - `crDroidAndroid-xxx_boot.img`
```
don't use ` as that renders as copy command and with - in front, is useless
can use some italic, bold or bold-italic

Terminator-J replied:
that's weird, should render as bullet points (unordered list), with inline-code format applied to entry after bullet.

i'll make it dumber.

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+  - MindTheGapps _should_ always be a good option (does not include PrivateCompute or SettingsIntelligence services):
 
-  MindTheGapps _should_ always be a good option, regardless of how much Pixel integration the crDroid dev team is including or not at the time:
+    Download **[MindTheGapps 14.0.0-arm64](https://github.com/MindTheGapps/14.0.0-arm64/releases)**
```
don't link all the gapps available
should state what you use and test with

also
link is already on download page, gapps button - no need to link here also (just state that)

Terminator-J replied:
I have always maintained a list of known-good Google Apps packages.
I always test both NikGapps "crdroid-official" and MindTheGapps (because it's maintained partially by LineageOS devs and if it isn't working, then that means our *ROM* is broken and should get fixed).
I became a maintainer because I actually wanted to run microG or even vanilla.

So these are all valid options that should work, and users should know that.
I'm open to suggestions for where to put that information so that it's visible, not hidden/buried.

gwolf2u replied:
then if you test and support all, I don't mind

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
-  - Let's save that kind of thing for after you've booted the first time & finished the Setup Wizard. Best to get a clean SafetyNet/Play Integrity attestation first, then reboot to recovery to flash root tools after.
+  - Wait until after first boot & finishing the Setup Wizard. Best to get a clean SafetyNet/Play Integrity attestation first, then reboot to recovery to flash root tools later on.
 
 - Disable_Dm-Verity_ForceEncrypt:
```
no need of it as we don't support it anyway

Terminator-J replied:
People use custom ROMs to be able to root.

It's good to tell users to wait until after first boot to try using Magisk, or flashing my KSU-enabled kernel zip, and has saved me support time to have that sentence there.

gwolf2u replied:
this was for dm-verity, not for root

Terminator-J replied:
I constantly get asked about it and wanted to specifically say "NO" in advance.

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
 
-Notice that each command is prefixed with a **./** This is important. Please keep that prefix when you run each command.
-_... unless you know how to add things to your $PATH and can verify which executable you're running, of course_
+**Notice that each command is prefixed with a `./`. This is important. Please keep that prefix when you run each command.**
```
don't use `

Terminator-J replied:
this is literally what inline code backtick markdown is for.

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+_(unless you know how to add things to your $PATH and can verify which executable is running, of course)_
 
-2. Move the following files into the extracted platform-tools directory (so that the ROM zip is in the same folder as adb & fastboot binaries, not one level above it or anything else):
+- Step 1: Unzip the `platform-tools-latest-xxx.zip` file you downloaded earlier to any directory you can run programs from.
```
don't use `

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
-    - crDroidAndroid-14.0-xxx_super_empty.img
-    - crDroidAndroid-14.0-xxx_vbmeta.img
-    - the latest OTA zip for you device (crDroidAndroid-14.0-xxx.zip)
+- Step 2: Move the following files into the extracted platform-tools directory (so that they are in the same directory level as the `adb` and `fastboot` executable files), and rename the `.img` files to lose the `crDroidAndroid-xxx` prefixes:
```
don't use `

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
 
-  - Optional addons:
-    - the Android 14.0 arm64 Google Apps or microG installer zip you decided on (if any)
+  - `crDroidAndroid-xxx_boot.img` renamed to `boot.img`
```
don't use `

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+  - (optional) the Google Apps or microG installer zip you decided on
 
-3. Open your terminal and navigate to the extracted platform-tools folder. Use **cd** to navigate to the folder.
+- Step 3: Open your terminal and `cd` to the extracted platform-tools directory.
```
don't use `

Terminator-J replied:
fine, i will use a generic verb "navigate" instead of shell command.

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+- Step 4: Boot phone to the OEM bootloader.
+
+  - Unplug USB cable from phone, power it off, then hold all three buttons (Vol Up + Vol Down + Power) until you see a screen with giant green text saying "Start" at the top of the screen.
+  - Verify you have correctly installed OnePlus USB drivers by plugging the USB cable back in, and running `fastboot devices` from the terminal window. You should see your device's serial number in the list of devices attached. If not, try reinstalling drivers, try a different cable, or a different USB port, or a USB hub, etc until it works.
```
don't use `

Terminator-J replied:
that's literally the command I want them to be able to copy & paste into the terminal.
this is what the backtick is for.

gwolf2u commented on 10/enchilada.md:
````diff
@@ -1,191 +1,202 @@
-    ./fastboot flash dtbo crDroidAndroid-14.0-xxx_dtbo.img
-    ./fastboot flash vbmeta crDroidAndroid-14.0-xxx_vbmeta.img
-    ```
+  - Use the volume buttons to page through the boot options until you see `Recovery mode`. Press the power button to select that option.
````
don't use `

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+  - Use the volume buttons to page through the boot options until you see `Recovery mode`. Press the power button to select that option.
 
-6. Erase the old Android partitions in the current slot with the following:
+- Step 7: In recovery, choose `Advanced` then `Enter fastboot` to enter fastbootd ("userspace fastboot"), and plug the USB cable back into phone.
```
don't use `

gwolf2u commented on 10/enchilada.md:
````diff
@@ -1,191 +1,202 @@
-   ./fastboot erase odm
-   ./fastboot erase vendor
-   ```
+  - Run `fastboot devices` in terminal to make sure that the userspace fastbootd drivers are working. If not, unplug & re-plug the USB cable again or check for driver updates via Device Manager or Windows Update.
````
don't use `

Terminator-J replied:
Again, this is the command I want them to copy & paste, it makes sense to have it inline.
This is what the backtick is for.

gwolf2u commented on 10/enchilada.md:
````diff
@@ -1,191 +1,202 @@
+  ```
 
-7. Unplug the USB cable & reboot your phone into recovery.
+  WARNING: If fastboot on the computer returns some error message about not recognizing `wipe-super` or prints a long help message instead, then that means you are either running a very old version of the fastboot executable or the phone was still in OEM bootloader instead of recovery-based fastbootd. Please go back and download the latest version of platform-tools as mentioned in the "Required f ...
````
don't use `

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+  WARNING: If fastboot on the computer returns some error message about not recognizing `wipe-super` or prints a long help message instead, then that means you are either running a very old version of the fastboot executable or the phone was still in OEM bootloader instead of recovery-based fastbootd. Please go back and download the latest version of platform-tools as mentioned in the "Required f ...
 
-  - Do this by using the volume buttons to scroll through the options until you see **Recovery mode**. Press the power button to select that option.
+- Step 9: Choose `Enter recovery` on phone to return to recovery, then choose `Apply update` then `Apply from ADB` so that it is waiting to receive an installation package from the computer.
```
don't use `

gwolf2u commented on 10/enchilada.md:
````diff
@@ -1,191 +1,202 @@
+  ```
 
-9. Initialize the retrofit super partitions for the current slot:
+  - During a successful sideload, the process will be paused at 47% on the computer, and on the phone you'll be prompted to reboot to recovery again in order to continue flashing Addons (like Google Apps or microG, since those are not built into crDroid). Tap on "Yes" to reboot into recovery (while `adb sideload` process on the computer will complete at 47%, possibly with an error message about " ...
````
don't use `

gwolf2u commented on 10/enchilada.md:
````diff
@@ -1,191 +1,202 @@
-    ```
-    ./fastboot wipe-super crDroidAndroid-14.x-xxx_super_empty.img
-    ```
+  If you encounter any other installation errors when sideloading besides ROM zip sideloading stopping at 47%, run `./adb pull /tmp/recovery.log` in terminal to get logs to share while asking for help.
````
don't use `

Terminator-J replied:
it's a shell command I want users to be able to copy.
that's what the inline code backtick is for.

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+  If you encounter any other installation errors when sideloading besides ROM zip sideloading stopping at 47%, run `./adb pull /tmp/recovery.log` in terminal to get logs to share while asking for help.
 
-  - When we sideload the crDroid ROM later, update_engine will handle initializing the other slot using the current slot's metadata.
+- Step 11: Back in recovery, choose `Apply update` then `Apply from ADB` again.
```
don't use `

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+  - Choose "Yes" again on phone when prompted to reboot to recovery.
 
-12. Run the following to sideload **copy_partitions.zip** to copy your firmware partitions to the other slot:
+- Step 13: If you want to run with Google Apps or microG since we don't ship them built-in, now is the time to install. On the phone, choose `Apply update` then `Apply from ADB` again, and from the terminal window on your computer, run the following (substituting the actual zip file name):
```
don't use `

gwolf2u commented on 10/enchilada.md:
````diff
@@ -1,191 +1,202 @@
+  ```
 
-  - You *may* receive an error from recovery, which will state the following:
+- Step 14: Since this is a clean install, navigate back to top level recovery menu on phone, and choose the `Factory reset` option & confirm in order to format the data partition & set up FBE encryption keys.
````
don't use `

gwolf2u commented on 10/enchilada.md:
````diff
@@ -1,191 +1,202 @@
-    > Signature verification failed  
-    > Install anyway?
-    ```
+- Step 15: Choose `Reboot system now` and remember: **𓂀 We love you 3000!**
````
don't use `- also remove "we love you 3000!" as while I understand why you use it, this is should be written in a pro mode without useless text

Terminator-J replied:
That message is literally my favorite thing about crDroid.
It has cheered me up over & over again, during some very dark times the last few years.

I would like to keep sharing that with my users, in case it helps some of them.

I will remove it so the official install instructions look more "pro", but I will also ask you to please reconsider in the future.

gwolf2u commented on 10/enchilada.md:
````diff
@@ -1,191 +1,202 @@
-    ```
-    ./adb sideload crDroidAndroid-14.0-xxx.zip
-    ```
+_Processes addon.d OTA survival scripts, so addons such as Google Apps / microG / Magisk that install survival scripts to `/system/addon.d/` should be preserved automatically_
````
don't use `

Terminator-J replied:
filenames and paths still seem like useful things to be able to quickly copy/paste, and using the markdown enables that.

i'm changing all of these things that aren't actual shell commands to double quotes instead, to match your preference; but it seems 1) messier, and 2) inelegant in comparison.

If the markdown parser for the crdroid.net site is always adding too much visual decoration when the inline code markdown is used, then that's a problem with the site parsing/styling, not a problem with the markdown being used to enhance readability/function.

can that extra decoration styling be fixed, so we can keep improved function?
much less error-prone than relying on users to be able to only select the text inside pairs of quotes (or some OSes trying to outsmart the user and "smart copy" all the wrong things).

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+_Processes addon.d OTA survival scripts, so addons such as Google Apps / microG / Magisk that install survival scripts to `/system/addon.d/` should be preserved automatically_
 
-  - After a successful sideload, the computer will show the process stopping at 47%, and on the phone you'll be prompted to reboot to recovery again in order to continue flashing Addons (like Google Apps or microG, since those are not built into crDroid). Choose "Yes" to reboot into recovery (and the **adb sideload** process on the computer will complete at 47%, possibly with an error message abo ...
+- Step 1: Open `Settings > System > System Updates` on the phone. Before loading an update to install, choose `Preferences` from the menu and turn on the `Prioritize update process` and `Delete updates when installed` toggles.
```
don't use `

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+  Protip: Use the "Caffeine" QS tile to keep the screen on for at least 10 minutes, or temporarily enable "Stay awake while charging" in Developer Options, or just temporarily set the Display timeout to 10 minutes to speed up the process. Installation slows down drastically if the screen turns off.
 
-15. If you want to run with Google Apps or microG since we don't ship them built-in, now is the time to install. On the phone, navigate to **Apply update -> Apply from ADB** again, and from the terminal window on your computer, run the following (substituting the actual zip file name):
+- Step 2: Now either check for updates, or choose `Local update` from the menu & select the desired ROM zip file from your phone's internal storage. Once the installer has verified the update, choose `Install` to begin installation.
```
don't use `

gwolf2u commented on 10/enchilada.md:
````diff
@@ -1,191 +1,202 @@
-    ```
-    ./adb sideload <Google Apps package>.zip
-    ```
+- Step 3: Reboot once installation has completed. If the screen has turned off or you navigated away from the `System updates` activity to use another app, then the only indication that updating has completed will be that the notification with progress bar has disappeared. Don't panic; just reboot the phone and you should see that the update has been applied.
````
don't use `

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+- Step 1: Make sure you have the **same Google Apps or microG installation zips** (or updated versions of the same Android version/architecture for you preferred package) as when you did the initial clean installation available to sideload. If you try to change to different GApps now, things will break.
 
-1. Open **Settings > System > Updater** on the phone. Before loading an update to install, choose **Preferences** from the menu and turn on the **Prioritize update process** and **Delete updates when installed** toggles. Now either check for updates, or choose **Local update** from the menu & select the desired ROM zip file from your phone's internal storage.
+- Step 2: Reboot phone to recovery and connect USB cable to computer once you're in the crDroid recovery environment. Open a terminal on the computer and run `adb devices` to verify that the phone is visible to the computer.
```
don't use `

Terminator-J replied:
it's a shell command I want them to be able to copy.

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+- Step 2: Reboot phone to recovery and connect USB cable to computer once you're in the crDroid recovery environment. Open a terminal on the computer and run `adb devices` to verify that the phone is visible to the computer.
 
-2. Once it has finished verifying the update, choose **Continue** to begin installation.
+- Step 3: On the phone, choose `Apply update` then `Apply from ADB` to begin the phone listening for an update. On the computer, run `./adb sideload path/to/crdroid.zip` (substituting the correct directory path & zip file name).
```
don't use `

Terminator-J replied:
Will update menu options with quotes, but keeping shell command as inline code.

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+- Step 3: On the phone, choose `Apply update` then `Apply from ADB` to begin the phone listening for an update. On the computer, run `./adb sideload path/to/crdroid.zip` (substituting the correct directory path & zip file name).
 
-_Protip: Enable "Stay awake while charging" in Developer Options, or use the "Caffeine" QS tile to keep the screen on, or just set the Display timeout to 10 minutes to cut the installation time from 2 hours down to less than 10 minutes. It slows down drastically if the screen goes off._
+- Step 4: On the computer, the process will pause at 47%, and on the phone you'll see it's prompting you to reboot to recovery to install addons (like Google Apps). Choose `Yes` on the phone to reboot to recovery now, and notice that the adb sideload process has finished on the computer at 47%, possibly with an error message about "no error". This is "normal". _Thanks, Google!_
```
don't use `

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+  DO NOT SKIP REBOOTING TO RECOVERY TO INSTALL ADDONS. This is due to how slot-switching works.
 
-1. Make sure you have the *same Google Apps or microG installation zips* (or updated versions of the same Android version/architecture) as when you did the initial clean installation available to sideload. If you change to different GApps now, things will get broken.
+- Step 5: If you have any addons to flash (Google Apps, microG, custom kernel zips, etc) now is the time to sideload them. On the phone, choose `Apply update` then `Apply from ADB` to start the phone listening, and then run `./adb sideload path/to/installation.zip` from the computer for each thing you need to install.
```
don't use `

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+  Again: Recovery does **not** process addon.d OTA survival scripts like the regular OTA updater in `Settings` does, so you CANNOT skip installing the Google Apps zip now if you previously had them installed, or it will ruin your current installation and you'll have to deal with either factory resetting or wiping runtime permissions.
 
-3. On the phone, choose **Apply update -> Apply from ADB** to begin the phone listening for an update. On the computer, run **adb sideload path/to/crdroid.zip** (substituting the correct path & filename).
+- Step 6: On the phone, choose `Reboot system now` and remember: **𓂀 We love you 3000!**
```
don't use ` and the 3000

gwolf2u replied:
still we do love 3000 ❤️

gwolf2u commented on 10/enchilada.md:
```diff
@@ -1,191 +1,202 @@
+_These instructions have been verified using both old/crusty and fresh installations of Windows 10, with both built-in PowerShell 1.x & 2.x and updated 7.x versions, on at least three different machines, including multiple combinations of AMD and Intel motherboards and various USB chipsets/hubs/ports/cables, as well as on a 2019 MacBook Pro (Intel) running MacOS Sonoma. If something isn't working ...
 
-6. On the phone, choose **Reboot system now**.
+_(Guide adapted with permission from the Evolution-X installation guide originally by jabashque and AnierinBliss, with extensive adaptation & updating over the years by Terminator_J)._
```
useless text - we don't advertise other roms on crDroid

Terminator-J replied:
I'm not advertising other ROMs; I'm giving attribution.

Do you strip out "Copyright 2016, The Android Open Source Project" from the top of license files, or do you add your own line after?

gwolf2u replied:
Can leave the name of the author, just don't want other rom name on crDroid site

Terminator-J replied:
Okay, so keep the people names and drop the mention of Evo-X by name?

gwolf2u reviewed (changes requested):
lots of changes there
remove useless text
don't use ` unless you use it for command with copy option
remove adverts

also did it only for enchilada as fajita is mainly the same and don't want to spend more time seeking all those out

Terminator-J commented:
Thank you for taking time to point out all the formatting and style things you want to see changed; that took a lot of work & time and I respect that.

Terminator-J commented:
Okay, personally: I hate dry, terse "documentation" that is just a list of steps that doesn't include any explanation or give any options or context or useful examples. That's why I started writing my own guides years ago, before we had "install_docs" available. People liked my guides & explanations on XDA forums. My users like it. It's something I'm very personally invested in, and as I'm sure you understand,... the only way people will keep volunteering for open-source projects is if they're personally invested in it and enjoy doing it.

Obviously you have a different idea of what you want documentation to look like for the official site.
I very much understand adapting the writing to make the site look consistent.

I will rewrite to make it a simple list of steps.

As a compromise, I would like to add a sentence at the end with a link to my XDA posts or GitHub Gists that says something like "For expanded installation guide, optional Google Apps/microG recommendations, and FAQs, click here" or something similar.

Would that be okay?

Terminator-J committed:
[SQUASHME] Revise OP6-series install_docs per review v1

Terminator-J commented:
Had to step away for a few days due to unrelated circumstances.

Rewrote to be more "just the pre-installation, first install, and update install steps" and less "start-to-finish guide", according to your notes:
- Removed all inline code markdown, and replaced with either double quotes for filenames/menu option names, or triple-backtick blocks for each terminal command
- Removed lots of other excess markdown to simplify layout/formatting
- Removed lots of extra "commentary" text, moved important notes/warnings to end of each step with consistent label
- Removed many extra links due to old "all in one" guide, directed back to "see ___ button on download page" instead
- Added two external links to streamline "Pre-installation" section instead of trying to solve all possible problems in advance
- Moved attribution line to bottom, removed advertising

Thank you for your patience!

gwolf2u commented:
this the final revision?

Terminator-J committed:
[SQUASHME] Revise OP6-series docs per review v2

Terminator-J commented:
Okay, should be ready to squash & merge!
Thank you so much for the guidance.
